src/Cli/dotnet/commands/dotnet-workload/install/IInstaller.cs

```diff
@@ -29,6 +29,8 @@ internal interface IInstaller : IWorkloadManifestInstaller
 
         void AdjustWorkloadSetInInstallState(SdkFeatureBand sdkFeatureBand, string workloadVersion);
 
+        void NotifyInstallComplete();
```

Comment:
Per our discussion, consider getting rid of this method and moving the garbage collection outside of the transaction.